core-data/reducer.js

```diff
@@ -82,8 +82,29 @@ export function postTypes( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing indexes data.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function indexData( state = {}, action ) {
+	switch ( action.type ) {
+		case 'RECEIVE_INDEX':
+			return {
+				...state,
+				...action.index,
+			};
+	}
+
+	return state;
+}
+
 export default combineReducers( {
 	terms,
 	media,
 	postTypes,
+	indexData,
```

Comment:
> Is `Data` suffix adding any value here?

For me personally, it denotes that it's the _data_ part of the index. `index` was too nebulous in my opinion (in comparison to `terms`, which more clearly communicates itself).

> Would we want to call other reducers `termsData`, `mediaData` as well for consistency's sake?

I don't think so. However, if you're not a fan of `indexData`, I'm open to renaming it. Alternatively, we could only put `themeSupports` in the store, instead of the entire index.